src/Core/Validation/FieldValidation/FieldValidationTrait.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use RuntimeException;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Validation\FieldValidation\FieldValidationInterface;
+use SilverStripe\Core\Validation\ValidationResult;
+
+/**
+ * Trait to add FieldValidator validation to a field, e.g. a DBField or FormField
+ * The field must implement FieldValidationInterface
+ */
+trait FieldValidationTrait
+{
+    use Configurable;
+
+    /**
+     * FieldValidators configuration for the field
+     *
+     * Each item in the array can be one of the following
+     * a) MyFieldValidator::class,
+     * b) MyFieldValidator::class => [null, 'getSomething'],
+     * c) MyFieldValidator::class => null,
+     *
+     * a) Will create a MyFieldValidator and pass the name and value of the field as args to the constructor
+     * b) Will create a MyFieldValidator and pass the name, value, and pass additional args, where each null values
+     *    will be passed as null, and non-null values will call a method on the field e.g. will pass null for the first
+     *    additional arg and call $field->getSomething() to get a value for the second additional arg
+     * c) Will disable a previously set MyFieldValidator. This is useful to disable a FieldValidator that was set
+     *    on a parent class
+     *
+     * You may only have a single instance of a given FieldValidator class per field, e.g. you can't have two
+     * instances of a `MyFieldValidator` class for the same field.
+     */
+    private static array $field_validators = [];
+
+    /**
+     * Validate this field using FieldValidators
+     */
+    public function validate(): ValidationResult
+    {
+        $result = ValidationResult::create();
+        $fieldValidators = $this->getFieldValidators();
+        foreach ($fieldValidators as $fieldValidator) {
+            $result->combineAnd($fieldValidator->validate());
+        }
+        return $result;
+    }
+
+    /**
+     * Get instantiated FieldValidators based on `field_validators` configuration
+     */
+    private function getFieldValidators(): array
+    {
+        $fieldValidators = [];
+        // Used to disable a validator that was previously set with an int index
+        if (!is_a($this, FieldValidationInterface::class)) {
+            $message = get_class($this) . ' must implement interface ' . FieldValidationInterface::class;
+            throw new RuntimeException($message);
+        }
+        /** @var FieldValidationInterface|Configurable $this */
```

Comment:
```suggestion
        /** @var FieldValidationInterface $this */
```
It's configurable 'cause this trait is configurable. Only need to hint the interface.

Reply:
My VSCode requires `|Configurable` or it'll complain about the signature, so leave it in.  I know that's fixing a shortcoming of VSCode and not our code, though given the purpose of these blocks are for typing hinting IDE's, pragmatically speaking it should stay